08-transport.Rmd

```diff
@@ -300,26 +322,34 @@ Desire lines contain only two vertices (their beginning and end points) but rout
-Remote routing services use an API to receive queries about origin and destination points, and then return the results over the internet.
+Routing is computationally intensive and might be slow on a local machine (depending on the hardware).
+In any case, local routing requires the route network to be stored on a local computer (we will see how in section \@ref(route-networks)).
+By contrast, remote routing services use a web API to receive queries about origin and destination points, run the routing on the route network on a powerful server and return just the results, the routes, to the user, the so-called client.
 There are many advantages of using remote routing services, including the fact that they can be up-to-date, have global coverage, and run on a set-up that was designed for the job, explaining this section's focus on online routing services.
-Before proceeding, however, you should be aware of the disadvantages of online routing services: they can be slow due to their reliance on internet connections and expensive.
+Before proceeding, however, you should be aware of the disadvantages of online routing services: they are slow or even inaccessible when accessed with a bad internet connection and can be expensive, i.e., not all services are free of charge.
+For instance, the Google routing API sets the limit to 2500 free queries per day ^[`ggmap::route()` lets you use Google's routing API through R.].
 <!-- Todo: add link to Mark's presentation on dodgr vs Google routing costs (RL) -->
 
-A popular and free (for limited uses) routing service is the Open Source Routing Machine ([OSRM](http://project-osrm.org/)).
-Instead of routing *all* desire lines generated in the previous section, which would be time and memory-consuming, we will focus attention on only the desire lines of policy interest.
+A popular and free (for limited uses) routing service is the Open Source Routing Machine ([OSRM](http://project-osrm.org/)).^[Another popular OSM routing API is https://openrouteservice.org/.]
+Instead of routing *all* desire lines generated in the previous section, which would be time and memory-consuming, we will focus on the desire lines of policy interest.
 The benefits of cycling trips are greatest when they replace car trips.
 Clearly not all car trips can realistically be replaced by cycling, but 5 km Euclidean distance (or around 6-8 km of route distance) is easily accessible within 30 minutes for most people.
 Based on this reasoning
 <!-- Todo: add references supporting this (RL) -->
-we will only route desire lines along which a high (300+) number of car trips take place that are up to 5 km in distance:
+we will only route desire lines along which a high (300+) number of car trips take place that are up to 5 km in distance.
 
+<!-- st_length works with sf 0.6.0 but not with 0.6.1 -->
```

Comment:
just a remark: on my computer `st_length` worked with sf 0.6.0 but not with 0.6.1